app/main.py

```diff
@@ -2,24 +2,75 @@
 
 
 class IntegerRange:
-    pass
+
+    def __set_name__(self, owner: any, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(self, instance: any, owner: any) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: any, value: any) -> None:
+        setattr(instance, self.protected_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: IntegerRange,
+                 weight: IntegerRange,
+                 height: IntegerRange) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    def __init__(self) -> None:
+        super().__init__(age=IntegerRange(4, 14),
+                         weight=IntegerRange(20, 50),
+                         height=IntegerRange(80, 120))
```

Comment:
```suggestion
        age = IntegerRange(4, 14),
        weight = IntegerRange(20, 50)
        height = IntegerRange(80, 120)
```
The same fix in AdultSlideLimitationValidator

Reply:
I don't understand, do I need to remove the constructor here or not?

Reply:
Yeah, you do not need constructor just write 
age = IntegerRange(4, 14)
 and others